pkg/service.go

```
@@ -0,0 +1,69 @@
+package pkg
+
+import (
+	"errors"
+	"fmt"
+	"time"
+	log "github.com/sirupsen/logrus"
+)
+
+
+type CoffeeService struct {
+	Redis *RedisRepository
+	Validator * QuotaValidator
+}
+
+func (cfs *CoffeeService) ProcessOrder(quota string, user_id string, newOrder string)(string,error){
+	if err := cfs.validateUserAndQuota(quota, user_id); err != nil {
+		return "",err
+	}
+
+	//validate user history 
+	history, _ := cfs.Redis.Find(user_id)
```

don't swallow errors please

---

fixed